packages/create/bin/create-stacks-workspace.ts

```diff
@@ -223,6 +223,14 @@ async function main(parsedArgv: yargs.Arguments<CreateStacksArguments>) {
     const targetDirectory = path.resolve(dir);
     const isTargetDirectoryCurrent = targetDirectory === process.cwd();
 
+    const isNxJson = findFile('nx.json', targetDirectory, 3);
+    if (!isNxJson) {
+        console.error(
+            chalk.red`Stacks workspace can only be created inside an existing Nx Monorepo. Please select as such and try again.`,
+        );
+        process.exit(1);
+    }
+
```

Comment:
This unfortunately won't be sufficient, running the following command adds Nx to a non-monorepo `npx nx@latest init` according to the Nx docks for "Installing Nx on a Non-Monorepo Project", this will subsequently create the nx.json file. 

While we still need to check for this file, I'd also recommend checking the repository structure by utilising getProjects() from @nrwl/devkit. In a non-monorepo this returns the parent folder itself, so theoretically, if you're able to call this function and check that the project returned from it is the root/parent folder of the repo, then you know it's not a monorepo.

Reply:
So the check will traverse up 3 levels to find an nx.json that will verify such a monorepo. I'm not sure we can use getProjects without the nx tree being present, which is the case if we run it here.

I didn't understand your first point paragraph btw 😅

Reply:
In fact there's multiple things to consider here:
1. RE my first paragraph, in short, I'm saying that the command `npx nx@latest init` adds Nx to any repository. Therefore we can't rely on just nx.json to prove if it's a monorepo, as it can be present in a non-monorepo too.
2. We're calling this function in the create-stacks-workspace script, at which point, the file would never exist, as this script is to create a brand new repo with stacks.
3. The location of this check needs to be in every single one of our plugins, not the create script. We want to ensure that users only use our plugins in an appropriate repo, therefore we could utilise the getProjects() function, as each generator for our plugins will have access to the tree